src/Backups/BackupImpl.cpp

```diff
@@ -166,6 +172,12 @@ void BackupImpl::open(const ContextPtr & context)
         params.backup_info = *base_backup_info;
         params.open_mode = OpenMode::READ;
         params.context = context;
+        /// use_same_s3_credentials_for_base_backup should be inherited for base backups
+        params.use_same_s3_credentials_for_base_backup = use_same_s3_credentials_for_base_backup;
+
+        if (params.use_same_s3_credentials_for_base_backup)
+            backup_info.copyS3CredentialsTo(params.backup_info);
```

Comment:
Probably it's better to modify `base_backup_info` here too so

```
BACKUP ... TO S3(url, credentials) SETTINGS base_backup=S3(base_url), use_same_s3_credentials_for_base_backup=true
```

will write `S3(base_url, credentials)` as the base backup to the metadata.

Reply:
Ok, updated.